include/dxc/Support/dxcapi.extval.h

```diff
@@ -1,8 +1,160 @@
 #include "dxc/Support/dxcapi.use.h"
+#include "dxc/WinAdapter.h"
+#include "llvm/Support/raw_ostream.h"
+
 #include <cassert>
 #include <string>
 
 namespace dxc {
+
+class ExternalValidationHelper3 : public IDxcCompiler3 {
+
+public:
+  CComPtr<IDxcCompiler3> m_pCompiler3;
+  CComPtr<IDxcValidator> m_pValidator;
+
+  ExternalValidationHelper3() {
+    m_pCompiler3 = nullptr;
+    m_pValidator = nullptr;
+  }
+
+  HRESULT STDMETHODCALLTYPE Compile(
+      _In_ const DxcBuffer *pSource, ///< Source text to compile.
+      _In_opt_count_(argCount)
+          LPCWSTR *pArguments, ///< Array of pointers to arguments.
+      _In_ UINT32 argCount,    ///< Number of arguments.
+      _In_opt_ IDxcIncludeHandler
+          *pIncludeHandler,  ///< user-provided interface to handle include
+                             ///< directives (optional).
+      _In_ REFIID riid,      ///< Interface ID for the result.
+      _Out_ LPVOID *ppResult ///< IDxcResult: status, buffer, and errors.
+      ) override;
+
+  HRESULT STDMETHODCALLTYPE Disassemble(
+      _In_ const DxcBuffer
+          *pObject,     ///< Program to disassemble: dxil container or bitcode.
+      _In_ REFIID riid, ///< Interface ID for the result.
+      _Out_ LPVOID *ppResult)
+      override ///< IDxcResult: status, disassembly text, and errors.
```

Comment:
The result comment is in the wrong place, but see my previous note about just removing these unnecessary comments.